File: src/core/parser.ts

import { readFile } from '../io/reader';
import { log } from '../io/logger';

export function parse(input: string) {
  const data = readFile(input);
  log('parsing');
  return tokenize(data);
}

export function tokenize(data: string) {
  log('tokenizing');
  return data.split(' ');
}

File: src/core/parser.c
#include "parser.h"
#include "../io/reader.h"
#include "../io/logger.h"

void parse(const char* input) {
    char* data = read_file(input);
    log_msg("parsing");
    tokenize(data);
}

void tokenize(const char* input) {
    log_msg("tokenizing");
}
